### DISCUSSION TITLE
How to specify different quadrature in different direction

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#23915] [Q&A General] [answered]
Hi

I have a sample 2D domain below

![image](https://user-images.githubusercontent.com/38023107/228971457-a6b36f7e-aba4-430f-ba02-84da1f29dfe8.png)

I want to use different quadrature rules in both x and y-directions. Specifically, I want to use full integration in the y-axis with fine discretization, and mid-point rule in the x-direction. How do I specify these directions?

Thank you.

**Comment by GiudGiud** [comment]
You ll want to build your own quadrature rule to do this, we dont have that one

**Reply by salaudeen-ya** [reply]
This has to be in the libmesh, right?

**Reply by GiudGiud** [reply]
yes

**Reply by salaudeen-ya** [reply]
I want to modify the way the `tensor_product_quad` function works in the quadrature.C below
https://github.com/libMesh/libmesh/blob/14633497ea867c56053d1ea7a1fbe0442727eb24/src/quadrature/quadrature.C

The idea is to have parameters `npx` and `npy` in the quadrature block. This `npx` and `npy` variables would help in determining if the quadrature tensor product is to be done only in one direction (for a 2D case), or for corners. The `npx` and `npy` variables would be integers.

I know I would have to populate the `tensor_product_quad` function in the quadrature.C file based on the values I inputed as `npx` and `npy` in the input file by including the variables in the header file:

https://github.com/libMesh/libmesh/blob/14633497ea867c56053d1ea7a1fbe0442727eb24/include/quadrature/quadrature.h

How best can I achieve this?

**Reply by GiudGiud** [reply, marked as answer]
Start by making a new quadrature class in libmesh.
You can define new weights. & q points but you can also override all these routines, including tensor_product_quad to make them behave differently for your quadrature

Once it s accepted, we can add it to the list in MOOSE
